src/forms/Typeahead.jsx

```diff
@@ -14,32 +14,91 @@ const defaultItemToString = itemValue => (typeof itemValue === 'string' ? itemVa
  * @module Typeahead
  */
 class Typeahead extends React.PureComponent {
+	constructor(props) {
+		super(props);
+
+		this.handleSelection = this.handleSelection.bind(this);
+		// this.handleSingleSelection = this.handleSingleSelection.bind(this);
+		// this.handleMultiselectSelect = this.handleMultiselectSelect.bind(this);
+		this.stateReducer = this.stateReducer.bind(this);
+	}
+
+	// handleMultiselectSelect(item) {
+	// 	this.props.onSelect(item, [...this.props.multiSelectValues, item]);
+	// }
+
+	// handleSingleSelection(selectedItem, stateAndHelpers) {
+	// 	!this.props.multiSelect && this.props.onSelect && this.props.onSelect(selectedItem, stateAndHelpers);
+	// }
+
+	handleSelection(selectedItem, stateAndHelpers) {
+		if(this.props.multiSelect) {
+			this.props.onSelect(selectedItem, [...this.props.multiSelectValues, selectedItem]);
+		} else {
+			this.props.onSelect && this.props.onSelect(selectedItem, stateAndHelpers);
+		}
+	}
+
+	stateReducer(state, changes) {
```

Comment:
This is super interesting. Is it so that a user can select using the mouse or keyboard?

Reply:
Users can select by mouse or keyboard by default, but this gives us more control over what gets passed to Downshift's internal state so we can do things like keep the menu open by default